test/airbrake/utils_test.exs

```diff
@@ -48,4 +46,76 @@ defmodule Airbrake.UtilsTest do
              }
     end
   end
+
+  describe "detuple/1" do
+    test "returns the original data when input is a struct with no tuples" do
+      input = %Struct{baz: 100, qux: 200}
+
+      assert Utils.detuple(input) == input
+    end
+
+    test "returns the original data when input is a map with no tuples" do
+      input = %{baz: 100, qux: 200}
+
+      assert Utils.detuple(input) == input
+    end
+
+    test "returns the original data when input is a list with no tuples" do
+      input = ["foo", "bar"]
+
+      assert Utils.detuple(input) == input
+    end
+
+    test "returns a list when input is a tuple" do
+      input = {:ok, "something"}
+
+      assert Utils.detuple(input) == [:ok, "something"]
+    end
+
+    test "returns the original value when the input is anything else" do
+      input = "something"
+
+      assert Utils.detuple(input) == input
+    end
+
+    test "returns de-tupled data when input contains deeply nested tuples" do
+      input = %Struct{baz: [1, 2, {3, 4}], qux: %{foo: %{bar: {9, 9, 9, 9}}}}
+      expected = %Struct{baz: [1, 2, [3, 4]], qux: %{foo: %{bar: [9, 9, 9, 9]}}}
+
+      assert Utils.detuple(input) == expected
+    end
+  end
+
+  describe "destruct/1" do
+    test "returns the original data when input is a map" do
+      input = %{baz: 100, qux: 200}
```

Comment:
Similarly, we can strengthen these tests:

```suggestion
      input = %{a: %Struct{baz: 100 qux: 200}, b: "foo", c: %Struct{baz: 1 qux: 2}}
```

Reply:
Good call. I added some improved tests in this commit, 2433bad5cfd03dab5d264f3a2c3a3d2f07986de2, let me know if you think we need more